### PR TITLE
API diff between .NET 7.0 and .NET 8 Preview 1

### PR DESCRIPTION
**Pull request by carlossanlop**
- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @dreddy-work @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @jeffhandley @karelz @ericstj @stephentoub

- Microsoft.VisualBasic: @jaredpar @cston @333fred
- Microsoft.Win32: @dotnet/area-microsoft-win32
- System.Buffers: @dotnet/area-system-buffers 
- System.CodeDom: @dotnet/area-system-codedom 
- System.Collections: @dotnet/area-system-collections 
- System.ComponentModel: @dotnet/area-system-componentmodel 
- System.ComponentModel.Composition: @dotnet/area-system-componentmodel-composition 
- System.Data: @ajcvickers @DavoudEshtehari @David-Engel @JRahnama 
- System.Diagnostics.Activity: @dotnet/area-system-diagnostics-activity 
- System.Diagnostics.CodeAnalysis: @AaronRobinsonMSFT 
- System.Diagnostics.EventLog: @dotnet/area-system-diagnostics-eventlog 
- System.Diagnostics.Metrics: @tommcdon @noahfalk 
- System.Diagnostics.PerformanceCounter: @dotnet/area-system-diagnostics-performancecounter 
- System.Diagnostics.Process: @dotnet/area-system-diagnostics-process 
- System.IO: @dotnet/area-system-io 
- System.Net: @dotnet/ncl 
- System.Numerics: @dotnet/area-system-numerics 
- System.Reflection: @dotnet/area-system-reflection 
- System.Reflection.Emit: @dotnet/area-system-reflection-emit 
- System.Reflection.Metadata: @dotnet/area-system-reflection-metadata 
- System.Resources: @dotnet/area-system-resources 
- System.Runtime.CompilerServices: @dotnet/area-system-runtime-compilerservices 
- System.Runtime.InteropServices: @dotnet/interop-contrib @lewing @pavelsavara 
- System.Runtime.Intrinsics: @dotnet/area-system-runtime-intrinsics 
- System.Runtime.Serialization: @HongGit @StephenMolloy 
- System.Runtime.Versioning: @buyaa-n 
- System.Runtime: @dotnet/area-system-runtime
- System.Security: @dotnet/area-system-security 
- System.Text: @dotnet/area-system-text-encoding 
- System.Text.Json: @dotnet/area-system-text-json 
- System.Text.RegularExpressions: @dotnet/area-system-text-regularexpressions 
- System.Threading: @mangod9 @kouvel 
- System.Threading.Tasks: @dotnet/area-system-threading-tasks 
- System.Timers: @dotnet/area-system-componentmodel 
- System.Transactions: @HongGit @roji
- System.Xml: @dotnet/area-system-xml 
- System.Web: @Rick-Anderson 
- System: @dotnet/area-system-runtime

### CODE REVIEW

**Review by dreddy-work — COMMENTED** [review]
WinForms looks good.

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Buffers.Binary.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# System.Buffers.Binary
+
+``` diff
+ namespace System.Buffers.Binary {
+     public static class BinaryPrimitives {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Buffers.Binary.md:5]: BinaryPrimitives LGTM

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Buffers.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# System.Buffers
+
+``` diff
+ namespace System.Buffers {
++    public static class IndexOfAnyValues {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Buffers.md:5]: IndexOfAnyValues LGTM.

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Numerics.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# System.Numerics
+
+``` diff
+ namespace System.Numerics {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Numerics.md:4]: System.Numerics LGTM

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.Arm.md [hunk 4]
```diff
@@ -0,0 +1,1328 @@
+# System.Runtime.Intrinsics.Arm
+
+``` diff
+ namespace System.Runtime.Intrinsics.Arm {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.Arm.md:4]: Arm intrinsics LGTM. This is jsut a lot of `ConstantExpected` attributes being added.

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.Wasm.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
+# System.Runtime.Intrinsics.Wasm
+
+``` diff
++namespace System.Runtime.Intrinsics.Wasm {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.Wasm.md:4]: Wasm intrinsics LGTM.

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.X86.md [hunk 6]
```diff
@@ -0,0 +1,519 @@
+# System.Runtime.Intrinsics.X86
+
+``` diff
+ namespace System.Runtime.Intrinsics.X86 {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.X86.md:4]: x86 intrinsics LGTM. Likewise a lot of new `ConstantExpected` attributes

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.md [hunk 7]
```diff
@@ -0,0 +1,441 @@
+# System.Runtime.Intrinsics
+
+``` diff
+ namespace System.Runtime.Intrinsics {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.Intrinsics.md:4]: Shared intrinsics LGTM. This is Vector512<T> and some additional APIs exposed for consistency.

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
+# System
+
+``` diff
+ namespace System {
```
Comment (tannergooding) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.md:4]: System APIs LGTM.

CC. @tarekgh for the DateTime API changes in your area.

**Review by tannergooding — APPROVED** [review]
Signed-off on all the areas I'm responsible for. Thanks!

**Review by lewing — COMMENTED** [review]
JSInterop changes lgtm (just [StringSyntaxAttribute("Json")] in appropriate places)

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.InteropServices.JavaScript.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# System.Runtime.InteropServices.JavaScript
+
+``` diff
+ namespace System.Runtime.InteropServices.JavaScript {
+     [CLSCompliantAttribute(false)]
+     [EditorBrowsableAttribute(EditorBrowsableState.Never)]
+     [SupportedOSPlatformAttribute("browser")]
+     public sealed class JSFunctionBinding {
+-        public JSFunctionBinding();
```
Comment (lewing) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.InteropServices.JavaScript.md:9]: Looking into this

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.Emit.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
+# System.Reflection.Emit
```
Comment (ericstj) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.Emit.md:1]: LGTM

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.Metadata.Ecma335.md [hunk 11]
```diff
@@ -0,0 +1,13 @@
+# System.Reflection.Metadata.Ecma335
```
Comment (ericstj) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.Metadata.Ecma335.md:1]: LGTM

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.PortableExecutable.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
+# System.Reflection.PortableExecutable
```
Comment (ericstj) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.PortableExecutable.md:1]: LGTM

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.md [hunk 13]
```diff
@@ -0,0 +1,10 @@
+# System.Reflection
```
Comment (ericstj) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Reflection.md:1]: LGTM

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.CompilerServices.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
+# System.Runtime.CompilerServices
```
Comment (ericstj) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.CompilerServices.md:1]: LGTM

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.InteropServices.JavaScript.md (hunk 9 above)
Comment (ericstj) [reply release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Runtime.InteropServices.JavaScript.md:9]: Seems to be caught with API Compat and suppressed: https://github.com/dotnet/runtime/blob/451d84e82f4f9c878f530c294d589802eb326da6/src/libraries/apicompat/ApiCompatBaseline.NetCoreAppLatestStable.xml#L6

This was the PR: https://github.com/dotnet/runtime/pull/76196

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Net.Quic.md [hunk 15]
```diff
@@ -0,0 +1,10 @@
+# System.Net.Quic
```
Comment (rzikm) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Net.Quic.md:1]: System.Net.Quic LGTM!

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Net.Security.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
+# System.Net.Security
```
Comment (rzikm) [inline release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Net.Security.md:1]: Changes themselves look okay, but the diff looks weird (no body for CipherSuitesPolicy)

**Review by adamsitnik — APPROVED** [review]
`System.Diagnostics.Process` LGTM, thank you!

File: release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Net.Security.md (hunk 16 above)
Comment (carlossanlop) [reply release-notes/8.0/8.0.1/api-diff/Microsoft.NETCore.App/8.0.1_System.Net.Security.md:1]: It's not the ideal way, but I guess it makes sense. The tool considers the addition of new attributes as a change on the API signature. In this case, the CipherSuitesPolicy class and the NegotiateClientCertificateAsync method both got new attributes.

**Pull request merged**